daemon/cmd/daemon_main.go

```diff
@@ -1587,76 +1638,48 @@ func registerDaemonHooks(params daemonParams) error {
 				k8s.SetClients(params.Clientset, params.Clientset.Slim(), params.Clientset, params.Clientset)
 			}
 
-			bootstrapStats.earlyInit.Start()
-			initEnv()
-			bootstrapStats.earlyInit.End(true)
+			d, restoredEndpoints, err := newDaemon(
+				ctx, cleaner,
+				WithDefaultEndpointManager(ctx, endpoint.CheckHealth),
+				params.Datapath,
+				params.Clientset)
+			if err != nil {
+				return fmt.Errorf("daemon creation failed: %w", err)
+			}
+			daemon = d
+
+			daemonResolver.Resolve(daemon)
 
 			// Start running the daemon in the background (blocks on API server's Serve()) to allow rest
 			// of the start hooks to run.
-			go runDaemon(ctx, cleaner, params.Shutdowner, params.Clientset)
+			if !option.Config.DryMode {
```

Comment:
Interesting! That means that the other stuff in `runDaemon` is no longer necessary for unit testing :+1: 

I noticed that the `runDaemon()` function still has some `!option.Config.DryMode` checks inside, so we can probably drop those since the check now occurs here?

Reply:
Removed the DryMode checks from runDaemon and initEnv.